app/src/main/java/com/capstone/warungpintar/ui/addproduct/AddProductInActivity.kt

```diff
@@ -1,19 +1,90 @@
 package com.capstone.warungpintar.ui.addproduct
 
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.provider.MediaStore
+import android.util.Log
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
 import com.capstone.warungpintar.R
 import com.capstone.warungpintar.databinding.ActivityAddProductInBinding
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.Text
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.TextRecognizerOptionsInterface
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
 
 class AddProductInActivity : AppCompatActivity() {
     private lateinit var binding: ActivityAddProductInBinding
-
+    private val cameraRequest = 1888
+    private lateinit var bitmap : Bitmap
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityAddProductInBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        binding.btnAddproductClose.setOnClickListener{
+        binding.btnAddproductClose.setOnClickListener {
             onBackPressed()
+
+
+        }
+        if (ContextCompat.checkSelfPermission(applicationContext, Manifest.permission.CAMERA)
+            == PackageManager.PERMISSION_DENIED
+        )
+            ActivityCompat.requestPermissions(
+                this,
+                arrayOf(Manifest.permission.CAMERA),
+                cameraRequest
+            )
+        binding.kodestockEditTextLayout.setOnClickListener {
+            val cameraIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+            startActivityForResult(cameraIntent, cameraRequest)
+        }
+
+    }
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == cameraRequest) {
+            bitmap = data?.extras?.get("data") as Bitmap
+            val image = InputImage.fromBitmap(bitmap, 0)
```

Comment:
data buat bitmp dari intentnya null

Error: java.lang.NullPointerException: null cannot be cast to non-null type android.graphics.Bitmap